main.go

```diff
@@ -0,0 +1,252 @@
+			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		}
+	})
+
+	http.HandleFunc("/urls", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == http.MethodGet {
+			getUrlShortsHandler(w, r, db)
+		} else {
+			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		}
+	})
+
+	// Start the HTTP server
+	port := "1337" // Change this to the desired port
+	log.Printf("Server listening on port %s...\n", port)
+	log.Fatal(http.ListenAndServe(":"+port, nil))
+}
+
+// openDB opens a connection to the PostgreSQL database
+func openDB() (*sql.DB, error) {
+	connStr := fmt.Sprintf("host=%s port=%d user=%s password=%s dbname=%s sslmode=disable", dbHost, dbPort, dbUser, dbPass, dbName)
+	return sql.Open("postgres", connStr)
+}
+
+// getUsersHandler retrieves all users from the Users table and returns them as a JSON response
+func getUsersHandler(w http.ResponseWriter, r *http.Request, db *sql.DB) {
+	rows, err := db.Query("SELECT * FROM users")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error fetching users: %v", err), http.StatusInternalServerError)
+		return
+	}
+	defer rows.Close()
+
+	var users []User
+	for rows.Next() {
+		var user User
+		err := rows.Scan(&user.ID, &user.Name, &user.Email, &user.BirthDate)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Error scanning users: %v", err), http.StatusInternalServerError)
+			return
+		}
+		users = append(users, user)
+	}
+
+	// Convert users slice to JSON and write the response
+	w.Header().Set("Content-Type", "application/json")
+	err = json.NewEncoder(w).Encode(users)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error encoding users to JSON: %v", err), http.StatusInternalServerError)
+		return
+	}
+}
+
+func addUserHandler(w http.ResponseWriter, r *http.Request, db *sql.DB) {
+	if r.Method != http.MethodPost {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	var user User
+	if err := json.NewDecoder(r.Body).Decode(&user); err != nil {
+		http.Error(w, fmt.Sprintf("Error decoding JSON request: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	query := "INSERT INTO users (name, email, birthdate) VALUES ($1, $2, $3) RETURNING user_id"
+	err := db.QueryRow(query, user.Name, user.Email, user.BirthDate).Scan(&user.ID)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error inserting user into database: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	err = json.NewEncoder(w).Encode(user)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error encoding response to JSON: %v", err), http.StatusInternalServerError)
+		return
+	}
+}
+
+func getUrlShortsHandler(w http.ResponseWriter, r *http.Request, db *sql.DB) {
+	rows, err := db.Query("SELECT * FROM url_shorts")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error fetching url_shorts: %v", err), http.StatusInternalServerError)
+		return
+	}
+	defer rows.Close()
+
+	var urlShorts []URLShort
+	for rows.Next() {
```

Comment:
If you extract the logic to get the shorts in a service you will be in a good position to write a nice unit test.

Reply:
Maybe after adding all the basic functionality i will try to refactor the code in a proper structure, probably something like this:
myapp/
  ├── cmd/
  │   └── main.go
  ├── internal/
  │   ├── models/
  │   │   └── user.go
  │   ├── repository/
  │   │   └── user_repository.go
  │   ├── service/
  │   │   └── user_service.go
  │   └── handler/
  │       └── user_handler.go
  ├── pkg/
  │   └── database/
  │       └── database.go
  ├── migrations/
  │   └── <your_migration_files>.sql
  ├── config/
  │   └── config.yaml
  ├── Dockerfile
  └── docker-compose.yml